ci: update supported go versions from 1.15-1.17 to 1.15-1.18

dkegel-fastly opened the pull request:
Currently shows one or two tinygo-test failures on windows with 1.18

dkegel-fastly committed:
ci: updated supported go versions from 1.15-1.17 to 1.16-1.18

dkegel-fastly committed:
gofmt exec_posix.go

aykevl commented on .circleci/config.yml:
```diff
@@ -101,21 +101,21 @@ commands:
 
 jobs:
-  test-llvm11-go115:
+  test-llvm11-go116:
```
IMHO we should either drop support for Go 1.15 entirely or continue to support in including in CI. We can for example test Go 1.15 and 1.17: it's unlikely that with those tests, Go 1.16 will somehow break.
(I don't have a strong preference, but unless it holds back some new features I'd be in favor of keeping Go 1.15 support for now).

dkegel-fastly replied:
I changed the 1.16 to 1.15 in circleci.

dkegel-fastly committed:
Smoke test 1.15 still until we drop support for it

dkegel-fastly committed:
Makefile: tinygo-test: skip compress/flate test on windows for now

Lets us move forward until
https://github.com/tinygo-org/tinygo/issues/2762
is fixed.

dkegel-fastly committed:
gofmt src/machine/machine_mimxrt1062_spi.go to please 1.18

dkegel-fastly committed:
.github: use proper name for Mac OS 11 (10.16 was unofficial; we could also use 10.15)

deadprogram commented:
Merging now so we can get closer to a release. Your patient work on this is essential @dkegel-fastly thank you!
